GVRf/Framework/framework/src/main/jni/engine/renderer/gl_renderer.cpp

```diff
@@ -230,13 +231,16 @@ void GLRenderer::renderShadowMap(RenderState& rstate, Camera* camera, GLuint fra
     GL(glClear(GL_DEPTH_BUFFER_BIT | GL_COLOR_BUFFER_BIT));
 
     for (auto it = render_data_vector.begin();
-            it != render_data_vector.end(); ++it) {
-        GL(renderRenderData(rstate, *it));
+         it != render_data_vector.end(); ++it) {
+        RenderData* rdata = *it;
+        if (rdata->cast_shadows()) {
+            GL(renderRenderData(rstate, rdata));
+        }
     }
-
     glBindFramebuffer(GL_READ_FRAMEBUFFER, readFbo);
     glBindFramebuffer(GL_DRAW_FRAMEBUFFER, drawFbo);
```

Comment:
@roshanch I added these some time ago. We can't leave the shadow fbo bound since daydream doesn't expect you are going to return control back to daydream with a different fbo bound.